tests/ccl_test_redshifts.py

```diff
@@ -0,0 +1,149 @@
+def check_redshift_numerical():
+    """
+    Compare the redshift functions to a high precision integral.
+    """
+    # Redshift input
+    z_lst = [0., 0.5, 1., 1.5, 2.]
+                
+     # p(z) function for dNdz_tomog
+    def pz1(z_ph, z_s, args):
+        return np.exp(- (z_ph - z_s)**2. / 2.)
+
+    # Lambda function p(z) function for dNdz_tomog
+    pz2 = lambda z_ph, z_s, args: np.exp(-(z_ph - z_s)**2. / 2.)
+    
+    # Set up a function equivalent to the PhotoZGaussian
+    def pz3(z_ph, z_s, sigz):
+		sig = sigz*(1.+ z_s)
+		return (np.exp(- (z_ph - z_s)**2. / 2. / sig**2) / np.sqrt(2.
+		        *np.pi) / sig)
+
+    # PhotoZFunction classes
+    PZ1 = ccl.PhotoZFunction(pz1)
+    PZ2 = ccl.PhotoZFunction(pz2)
+    PZ3 = ccl.PhotoZGaussian(sigma_z0=0.1)
+    
+    # dNdz (in terms of true redshift) function for dNdz_tomog
+    def dndz1(z, args):
+        return z**1.24 * np.exp(- (z / 0.51)**1.01)
+    
+    # dNdzFunction classes
+    dNdZ1 = ccl.dNdzFunction(dndz1)
+    dNdZ2 = ccl.dNdzSmail(alpha = 1.24, beta = 1.01, z0 = 0.51)
+    
+    # Do the integral in question directly in numpy at high precision
+    zmin = 0.
+    zmax = 1.
+    zp = np.linspace(zmin, zmax, 1e4)
+    zs = np.linspace(0., 5., 1e4) # Assume any dNdz does not extend 
+    # above z=5
+    denom_zp_1 =np.asarray([np.trapz(pz1(zp, z, []), zp) for z in zs])
+    denom_zp_2 =np.asarray([np.trapz(pz2(zp, z, []), zp) for z in zs])
+    denom_zp_3 =np.asarray([np.trapz(pz3(zp, z, 0.1), zp) for z in zs])
+    np_dndz_1 = ([ dndz1(z, []) * np.trapz(pz1(zp, z, []), zp) / 
+                 np.trapz(dndz1(zs, []) * denom_zp_1, zs) for z in 
+                 z_lst])
+    np_dndz_2 = ([ dndz1(z, []) * np.trapz(pz2(zp, z, []), zp) / 
+                 np.trapz(dndz1(zs, []) * denom_zp_2, zs) for z in 
+                 z_lst])
+    np_dndz_3 = ([ dndz1(z, []) * np.trapz(pz3(zp, z, 0.1), zp) / 
+                 np.trapz(dndz1(zs, []) * denom_zp_3, zs) for z in 
+                 z_lst])
+    
+    # Check that for the analytic case introduced above, we get the 
+    # correct value.
+    for i in range(0, len(z_lst)):
+		assert_allclose(ccl.dNdz_tomog(z_lst[i], zmin, zmax, PZ1, 
+		               dNdZ1), np_dndz_1[i], rtol=TOLERANCE) 
+		assert_allclose(ccl.dNdz_tomog(z_lst[i], zmin, zmax, PZ1, 
+		               dNdZ2), np_dndz_1[i], rtol=TOLERANCE)  
+		assert_allclose(ccl.dNdz_tomog(z_lst[i], zmin, zmax, PZ2, 
+		               dNdZ1), np_dndz_2[i], rtol=TOLERANCE) 
+		assert_allclose(ccl.dNdz_tomog(z_lst[i], zmin, zmax, PZ2, 
+		               dNdZ2), np_dndz_2[i], rtol=TOLERANCE)  
+		assert_allclose(ccl.dNdz_tomog(z_lst[i], zmin, zmax, PZ3, 
+		               dNdZ1), np_dndz_3[i], rtol=TOLERANCE) 
+		assert_allclose(ccl.dNdz_tomog(z_lst[i], zmin, zmax, PZ3, 
+		               dNdZ2), np_dndz_3[i], rtol=TOLERANCE)  
+
+def test_redshift_analytic():
+    """
+    Test dNdz_tomog values against analytic toy example.
+    """
+
+    check_redshift_analytic()
```

Comment:
Other places use this style because the `check` function takes an argument, and you want to run it for a few different values of this argument.  Here, you could probably just rename the `check` functions as `test` functions and remove this level of indirection.

Reply:
Good point, will do.